### ISSUE TITLE
dialyxir not working for simple project

### ISSUE THREAD

**Issue by Fl4m3Ph03n1x** [jeremyjh/dialyxir#273]
### Environment

* Elixir & Erlang/OTP versions (elixir --version):

output of `asdf current`
```
elixir         1.7.3-otp-21(set by /home/fl4m3/Workspace/elixir_and_otp_guidebook/dialyxyr_playground/.tool-versions)
erlang         21.1    (set by /home/fl4m3/.tool-versions)
```

* Which version of Dialyxir are you using? (cat mix.lock | grep dialyxir):

`[{:dialyxir, "~> 0.5", only: [:dev], runtime: false}]`

### Current behavior

I have created a small project to play around with dialyxir. The experience so far has been puzzling at best.

Steps taken:
1. `mix new dialyxir_playground`
2. `cd dialyxir_playground`
3. `asdf local elixir 1.7.3-otp-21`
4. add dependency `[{:dialyxir, "~> 0.5", only: [:dev], runtime: false}]`
5. `mix do deps.get, deps.compile`

This is where I get the first issue. A ton of warnings. I am not sure if this is normal, but it sure doesn't give me a lot of confidence.

```
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  dialyxir 0.5.1
All dependencies up to date
==> dialyxir
Compiling 5 files (.ex)
warning: redefining module Dialyxir.Output (current version loaded from /home/fl4m3/.asdf/installs/elixir/1.7.3-otp-21/.mix/archives/dialyxir-1.0.0-rc.4/dialyxir-1.0.0-rc.4/ebin/Elixir.Dialyxir.Output.beam)
  lib/dialyxir/output.ex:1

warning: redefining module Dialyxir.Plt (current version loaded from /home/fl4m3/.asdf/installs/elixir/1.7.3-otp-21/.mix/archives/dialyxir-1.0.0-rc.4/dialyxir-1.0.0-rc.4/ebin/Elixir.Dialyxir.Plt.beam)
  lib/dialyxir/plt.ex:5

warning: redefining module Dialyxir.Dialyzer (current version loaded from /home/fl4m3/.asdf/installs/elixir/1.7.3-otp-21/.mix/archives/dialyxir-1.0.0-rc.4/dialyxir-1.0.0-rc.4/ebin/Elixir.Dialyxir.Dialyzer.beam)
  lib/dialyxir/dialyzer.ex:1

warning: redefining module Dialyxir.Project (current version loaded from /home/fl4m3/.asdf/installs/elixir/1.7.3-otp-21/.mix/archives/dialyxir-1.0.0-rc.4/dialyxir-1.0.0-rc.4/ebin/Elixir.Dialyxir.Project.beam)
  lib/dialyxir/project.ex:1

warning: redefining module Dialyxir.Dialyzer.Runner (current version loaded from /home/fl4m3/.asdf/installs/elixir/1.7.3-otp-21/.mix/archives/dialyxir-1.0.0-rc.4/dialyxir-1.0.0-rc.4/ebin/Elixir.Dialyxir.Dialyzer.Runner.beam)
  lib/dialyxir/dialyzer.ex:6

warning: redefining module Mix.Tasks.Dialyzer (current version loaded from /home/fl4m3/.asdf/installs/elixir/1.7.3-otp-21/.mix/archives/dialyxir-1.0.0-rc.4/dialyxir-1.0.0-rc.4/ebin/Elixir.Mix.Tasks.Dialyzer.beam)
  lib/mix/tasks/dialyzer.ex:1

Generated dialyxir app
```

But it doesn't stop here, here are more steps:

6. Add the bug test files into `lib`
7. `mix dialyzer`

And it explodes:

```
Compiling 5 files (.ex)
Generated dialyxir_playground app
** (UndefinedFunctionError) function Dialyxir.Output.info/1 is undefined or private
    Dialyxir.Output.info("Finding suitable PLTs")
    lib/mix/tasks/dialyzer.ex:158: Mix.Tasks.Dialyzer.run/1
    (mix) lib/mix/task.ex:316: Mix.Task.run_task/3
    (mix) lib/mix/cli.ex:79: Mix.CLI.run_task/2

```

### Expected behavior

It should build the PLT tables without blowing up and then analyze the buggy files.

**Comment by NobbZ** [comment]
This is because you have a mismatch of installed versions.

Please remove the global installation of dialyxir.

**Comment by Fl4m3Ph03n1x** [comment]
Just did that but the outcome doesn't change. Looks like `archive.uninstall` is not doing it's job.
What files do I have to delete manually in order to purge dialyxir from a machine?

**Comment by Fl4m3Ph03n1x** [comment]
Fixed. Turns out I had a lingering version of dialyxir isntalled globally with asdf `elixir 1.7.3-otp-21`. I had to go to `~/.asdf/installs/elixir/1.7.3-otp-21/.mix/archives` and manually delete the dialyxir folder. 

Once I did that, everything worked out perfectly for both versions.
Will close this issue and update the forum discussion. Should probably also update the REDME...

Thanks for the help.
